src/main/java/bio/terra/service/filedata/azure/tables/TableDependencyDao.java

```diff
@@ -0,0 +1,107 @@
+package bio.terra.service.filedata.azure.tables;
+
+import bio.terra.service.filedata.google.firestore.FireStoreDependency;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import com.azure.data.tables.models.TableTransactionAction;
+import com.azure.data.tables.models.TableTransactionActionType;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.ListUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TableDependencyDao {
+  private final Logger logger = LoggerFactory.getLogger(TableDependencyDao.class);
+  private static final String DEPENDENCY_TABLE_NAME_SUFFIX = "dependencies";
+  private static final int MAX_FILTER_CLAUSES = 15;
+
+  @Autowired
+  public TableDependencyDao() {}
+
+  @VisibleForTesting
+  public String getDatasetDependencyTableName(UUID datasetId) {
+    return "datarepo" + datasetId.toString().replaceAll("-", "") + DEPENDENCY_TABLE_NAME_SUFFIX;
+  }
```

Comment:
I just wrote up a shared utility method to do this sort of name creation so that we can share between the load history table  and these snapshot tables. I hope to get your feedback on this and then I can add this to the shared method in my PR!

Reply:
Yeah that sounds great!